frontend/src/context/store.ts

```diff
@@ -38,7 +38,15 @@ export const store = configureStore({
     // serialize the state
     serializableCheck: false,
     immutableCheck: {
-      ignoredPaths: ['mapState.layersData', 'analysisResultState.result'],
+      // do not check the following state branches for accidental
+      // mutations. This saves a lot of time in dev mode (but has no
+      // impact on production builds).
+      ignoredPaths: [
+        'mapState.boundaryRelations',
+        'mapState.layersData',
+        'analysisResultState.result',
+        'serverState.availableDates',
```

Comment:
Adding this line here saves enormous amounts of time in dev mode, both during initial loading and pretty much any interaction afterwards (for instance, a click on the bottom date selector takes about 1.5s vs 6-7s before).
`redux-toolkit` and its immutable middleware (in dev mode) runs all `redux` actions **twice**, and each time does a deep comparison of the entire state tree to check for accidental mutations. The `ignoredPaths` tells it to skip these parts of the tree. `serverStates.availableDates` holds around 80 layers of 250-3300 dates each, which won't actually change once they've been loaded, so this saves a ton of useless work.

[This](https://github.com/reduxjs/redux-toolkit/blob/b283b1c9d7fc9b7b9bd5729998ab84490db1dbe5/packages/toolkit/src/immutableStateInvariantMiddleware.ts#L166) is the checks it runs (for educational purposes) :wink: 

The more radical option is to set `immutableCheck: false` which entirely disables these checks. At that point, dev mode is virtually as fast as the production build, but the drawback is that we lose some useful checks.

A middle ground might be to disable the checks in dev mode, but turn them on in CI, as any mutation will throw an error. This obviously assumes that our tests cover all such cases.